discovery/v1.js

```diff
@@ -255,14 +255,30 @@ DiscoveryV1.prototype.addDocument = function(params, callback) {
   params = params || {};
 
   var query_params = pick(params, ['configuration_id'])
+  var formDataParams = pick(params, ['file', 'metadata']);
+
+  // if we get a buffer or object, we need to include stuff about filename for the service
+  if (formDataParams.file) {
+    if (typeof formDataParams.file.filename !== 'string' &&
+          !(formDataParams.file.options &&
+             typeof formDataParams.file.options.filename !== 'string') &&
+          !(formDataParams.file.path &&
+            typeof formDataParams.file.path !== 'string') &&
+          !(formDataParams.file.name &&
+            typeof formDataParams.file.name !== 'string')) {
```

Comment:
Man, I feel like I've helped to create a monstrosity.... 

We could probably just check for truthyness and not worry about the typeof check - .toString() works on prettymuch everything. That would save us a couple of lines here.